server/http/controller.go

```diff
@@ -484,9 +484,38 @@ func (c *controller) ExecuteDefinition(ctx context.Context, testDefinition opena
 		return c.executeTest(ctx, test.Model(), testDefinition.RunInformation)
 	}
 
+	if environment, err := def.Environment(); err == nil {
+		return c.createEnvFromDefinition(ctx, environment)
```

Comment:
This is returning the wrong response. It should return the testId, not te env ID.

Reply:
I'm a little confused here, when creating an environment from a definition file there won't be any info regarding the test so my idea was to return the environment id as the creation output, if that's not expected, do you think I should add a new field to the response to be environment id instead?

Reply:
Maybe this shouldn't be here then. Maybe the CLI should create the env before running the test and we only reference the environment here. Another thing, even if we leave this here, shouldn't this happen before the `executeTest` function call?

To execute a test we still need the transaction to exist.

Reply:
Oh I see what you did there. It is only be executed if the resource is an environment. Ignore my comments. Makes sense.